src/pinned_memory_manager.cc

```diff
@@ -153,6 +155,13 @@ PinnedMemoryManager::AllocInternal(
     if (status.IsOk()) {
       auto res = memory_info_.emplace(
           *ptr, std::make_pair(is_pinned, pinned_memory_buffer));
+
+      if (is_pinned) {
+        std::lock_guard<std::mutex> lk(alloc_info_mtx_);
+        used_pinned_memory_byte_size_ += size;
+        allocated_memory_info_.emplace(*ptr, size);
```

Comment:
Shouldn't we account for the pinned memory de-allocation performed in L180?
as in decrement and remove the used_pinned_memory_byte_size?

Reply:
How about moving the used_pinned_memory_byte_size within PinnedMemory class?
Then we can replace
```
*ptr = pinned_memory_buffer->managed_pinned_memory_.allocate(
        size, std::nothrow_t{});
```
and
```
pinned_memory_buffer->managed_pinned_memory_.deallocate(*ptr);
```
with `Allocate` and `Deallocate` calls respectively. These calls can internally keep record of how much memory is in use by this PinnedMemory object. 
I believe this should make the code more maintainable.